File: 1304.find-n-unique-integers-sum-up-to-zero.py
#
# @lc app=leetcode id=1304 lang=python
#
# [1304] Find N Unique Integers Sum up to Zero
#

# @lc code=start
class Solution(object):
    def sumZero(self, n):
        """
        :type n: int
        :rtype: List[int]
        """

        result = []
        for i in range(1, n // 2 + 1):
            result.append(i)
            result.append(-i)

        if n % 2 == 1:
            result.append(0)

        return result

# @lc code=end

File: 1304.find-n-unique-integers-sum-up-to-zero.js
/*
 * @lc app=leetcode id=1304 lang=javascript
 *
 * [1304] Find N Unique Integers Sum up to Zero
 */

// @lc code=start
/**
 * @param {number} n
 * @return {number[]}
 */
var sumZero = function (n) {
    const result = [];
    if (n % 2 !== 0) {
        result.push(0);
    }
    for (let i = 1; i <= n / 2; i++) {
        result.push(i);
        result.push(-i);
    }
    return result;
};
// @lc code=end
